test/framework/src/main/java/org/elasticsearch/test/TransportVersionUtils.java

```diff
@@ -54,15 +53,15 @@ public static TransportVersion randomVersionBetween(
     ) {
         int minVersionIndex = 0;
         if (minVersion != null) {
-            minVersionIndex = ALL_VERSIONS.indexOf(minVersion);
+            minVersionIndex = Collections.binarySearch(ALL_VERSIONS, minVersion);
         }
         int maxVersionIndex = ALL_VERSIONS.size() - 1;
         if (maxVersion != null) {
-            maxVersionIndex = ALL_VERSIONS.indexOf(maxVersion);
+            maxVersionIndex = Collections.binarySearch(ALL_VERSIONS, maxVersion);
         }
-        if (minVersionIndex == -1) {
+        if (minVersionIndex < 0) {
             throw new IllegalArgumentException("minVersion [" + minVersion + "] does not exist.");
-        } else if (maxVersionIndex == -1) {
+        } else if (maxVersionIndex < 0) {
             throw new IllegalArgumentException("maxVersion [" + maxVersion + "] does not exist.");
         } else if (minVersionIndex > maxVersionIndex) {
             throw new IllegalArgumentException("maxVersion [" + maxVersion + "] cannot be less than minVersion [" + minVersion + "]");
```

Comment:
since ALL_VERSIONS is sorted, this should be impossible, maybe change this to an AssertionError?

Reply:
It might be so if minVersion and maxVersion are specified the wrong way round by callers

Reply:
OK, changed